### PR TITLE
Convert showcase to submodule

### PR DESCRIPTION
**Pull request by joebochill**
For PXBLUE-XXXX.

#### Changes proposed in this Pull Request:
- Replace local showcase demo with a submodule for react-native-showcase-demo
- Update the scripts

### CODE REVIEW

**Comment by huayunh** [comment]
Code seems fine, will do a fresh clone later

File: package.json [hunk 1]
```diff
@@ -30,20 +31,21 @@
         "install:all-android": "cd components && yarn install && cd .. && yarn install:showcase-android && yarn install:storybook-android && yarn install:storybook-api",
         "link:components": "bash ./scripts/linkComponents.sh",
         "start": "yarn start:showcase",
-        "start:showcase": "yarn start:showcase-ios",
-        "start:showcase-ios": "yarn install:showcase-ios && cd demos/showcase && yarn ios",
-        "start:showcase-android": "yarn install:showcase-android && cd demos/showcase && yarn android",
+        "start:showcase": "yarn initialize && yarn start:showcase-ios",
```
Comment (huayunh) [inline package.json:34]: by running `yarn start:showcase`, the user is initializing the submodule twice

**Review by huayunh — COMMENTED** [review]
When I `yarn start:showcase`, I got 
```
...
$ curl --compressed -o- -L https://yarnpkg.com/install.sh | bash
Auto-linking React Native modules for target `reactNativeShowcaseDemo`: RNCMaskedView, RNGestureHandler, RNReanimated, RNSVG, RNScreens, RNVectorIcons, react-native-restart, and react-native-safe-area-context
Analyzing dependencies
Fetching podspec for `DoubleConversion` from `../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec`
Fetching podspec for `Folly` from `../node_modules/react-native/third-party-podspecs/Folly.podspec`
Fetching podspec for `glog` from `../node_modules/react-native/third-party-podspecs/glog.podspec`
[!] CocoaPods could not find compatible versions for pod "FlipperKit/FBDefines":
  In snapshot (Podfile.lock):
    FlipperKit/FBDefines (= 0.54.0, ~> 0.54.0)

  In Podfile:
    FlipperKit/FBDefines (~> 0.54.0)

None of your spec sources contain a spec satisfying the dependencies: `FlipperKit/FBDefines (~> 0.54.0), FlipperKit/FBDefines (= 0.54.0, ~> 0.54.0)`.

You have either:
 * out-of-date source repos which you can update with `pod repo update` or with `pod install --repo-update`.
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.
error Command failed with exit code 31.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 31.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 31.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I was running yarn start:showcase-android. It looks like the command does not automatically open up the tunnel page in the browser?

Otherwise everything spins ok.

**Comment by joebochill** [comment]
> When I `yarn start:showcase`, I got
> [ERROR MESSAGE OMITTED]
> 
> I then tried `yarn start:showcase-android`. It looks like the command does not automatically open up the tunnel page in the browser?
> 
> Otherwise everything spins ok.

Looks like you need to update your cocoapods version (I'm on 1.9.3...1.10 just came out a few weeks ago) and possibly your source repos as it mentions in the message. When I do a fresh clone, the pods install without any issues.

There is no browser tunnel page for this project - it's a RNC CLI project, not a Expo project. So it should just spin up the Metro server in a new terminal window and then start the app.

File: package.json (hunk 1 above)
Comment (joebochill) [reply package.json]: It shouldn't hurt (the second time it just runs prints the output saying it's linked without doing any actions), but I moved the initialization to the install script to be cleaner, so it should only happen once now.

**Comment by huayunh** [comment]
I am also on pod version 1.9.3. Let me try again.

**Review by huayunh — APPROVED** [review]
Works now. Nice!
